### PR TITLE
Clarify difference TFile::Open and new TFile() when file does not exist

### PR DESCRIPTION
**Pull request by ferdymercury**
# This Pull request:

## Changes or fixes:
Clarify how to check for non existing file when using TFile Open

## Checklist:

- [x] updated the docs (if necessary)

### CODE REVIEW

File: io/io/src/TFile.cxx
Comment (eguiraud) [inline io/io/src/TFile.cxx:4007]:
```suggestion
/// A nullptr is returned if the file does not exist.
```

**Comment by eguiraud** [comment]
Thanks @ferdymercury , looks good to me, but let's wait for a final confirmation by @pcanal before merging.

File: io/io/src/TFile.cxx
Comment (Axel-Naumann) [inline io/io/src/TFile.cxx:4007]:
```suggestion
/// A nullptr is returned if the file cannot be opened.
```
"if the file does not exist" sounds like `TFile::Open("foo.root", "RECREATE")` returns `nullptr` which is incorrect.

Comment (ferdymercury) [reply io/io/src/TFile.cxx:4007]: Why incorrect? This is what I see:

```
root [0] TFile* abc = TFile::Open("/tmp/blabla.root")
Error in <TFile::TFile>: file /tmp/blabla.root does not exist
(TFile *) nullptr

```

Comment (ferdymercury) [reply io/io/src/TFile.cxx]: I specified now the different behavior when reading or writing.

File: io/io/src/TFile.cxx
Comment (Axel-Naumann) [inline io/io/src/TFile.cxx:4008]: I think this is enough?

```suggestion
/// In READ mode, a nullptr is returned if the file does not exist or cannot be opened.
```
or even
```suggestion
/// In READ mode, a nullptr is returned if the file cannot be opened.
```

File: io/io/src/TFile.cxx
Comment (Axel-Naumann) [inline io/io/src/TFile.cxx:4009]:
```suggestion
/// In CREATE / RECREATE mode, a nullptr is returned if the file can not be created.
```

File: io/io/src/TFile.cxx
Comment (Axel-Naumann) [inline io/io/src/TFile.cxx:4007]: Please mention all the other options, too - or simplify / generalize as
```suggestion
/// A nullptr is returned if the file cannot be opened.
```

File: io/io/src/TFile.cxx
Comment (Axel-Naumann) [inline io/io/src/TFile.cxx:4010]:
```suggestion
/// In UPDATE mode, a nullptr is returned if the file can not be created or opened.
```

File: io/io/src/TFile.cxx
Comment (Axel-Naumann) [inline io/io/src/TFile.cxx:4008]:
```suggestion
/// In CREATE mode, a nullptr is returned if the file already exists or cannot be created.
```

**Review by Axel-Naumann — APPROVED** [review]
Thanks! Tiny improvements, waiting for @pcanal to merge!

**Pull request merged**
